### ISSUE TITLE
Release v1.2.0

### ISSUE THREAD

**Issue by treerink** [EC-Earth/ece2cmor3#498]
Overview of open issues: we need to decide what we include and maybe more important what not and list it:

**1. ua100m & va100m #496**
It seems better to retrieve `ua100m` & `va100m` from the grib codes `246.228` & `247.228,` the `ua-va-100m` branch has been created for this. However backward compatibility is hard, besides the `ifspar.json` it modifies all the ppt files in the control  output files quite a bit. So we propose not to merge this before the `EC-Earth3.3.1.1 -- ece2cmor3 v1.2.0` release, but thereafter, so this release set can be always used for the current situation.

**2. Ofx/sftof varies with time #483**
Does not have to hold the `EC-Earth3.3.1.1 -- ece2cmor3 v1.2.0` release, as it handles a Ofx variable.

**3. Missing time bounds #472**
`3hr` `tslsi`
`SImon` `sidivvel`
It would be really nice if this is fixed in ece2cmor3 v1.2.0. Does not hold the `EC-Earth3.3.1.1` release.

**4. QA-DKRZ only invalid values #471**
Note these variables have been discussed in [ece portal issue 609](https://dev.ec-earth.org/issues/609), though this issue should be the starting point to continue on this variables, the discussions over there are not clearly closed yet. At the moment of that discussion the preference policy wasn't set. In the mean while it is: preferences are applied at the level of the json data request file and not at the level of ppt, file_def & instruction files. With this fact actually a wrap up of 609 should be made, but this will be some work.

 | Table| variable    | showing                           | action | consensus | issue           |
 | ---- | ----------- | --------------------------------- | ------ | --------- | --------------- |
 | Omon | hflso       | only invalid values               | remove | yes       | [#609-17](https://dev.ec-earth.org/issues/609#note-17) [#609-18](https://dev.ec-earth.org/issues/609#note-18)|
 | Omon | hfsso       | only invalid values               | remove | yes       | [#609-17](https://dev.ec-earth.org/issues/609#note-17) [#609-18](https://dev.ec-earth.org/issues/609#note-18)|
 | Omon | rlntds      | only invalid values               | remove | yes       | [#609-17](https://dev.ec-earth.org/issues/609#note-17) [#609-18](https://dev.ec-earth.org/issues/609#note-18)|
 | Omon | agessc      | only invalid values               | OMIP   |           | [#609-19](https://dev.ec-earth.org/issues/609#note-19)                                                       |
 | Omon | hfibthermds | only invalid values               | remove | yes       | [#609-8 ](https://dev.ec-earth.org/issues/609#note-8 )                                                       |
 | Omon | ficeberg    | only invalid values               | remove | yes       | [#609-8 ](https://dev.ec-earth.org/issues/609#note-8 )                                                       |
 | Omon | hfcorr      | zero values on ocean rest invalid |        |           | [#534-7 ](https://dev.ec-earth.org/issues/534#note-7 )                                                       |
 | Omon | wfcorr      | zero values on ocean rest invalid |        |           | [#534-7 ](https://dev.ec-earth.org/issues/534#note-7 )                                                       |

Here we can either
1. Keep as it is and just remove those fields from the cmorised output
2. Remove these 8 variables from the json data request file, this asks for a `genecec` modification which affects the  json data request files in the `r6874-control-output-files` branch and thus holds the `EC-Earth3.3.1.1` release. Can be done in within one day.
3.  Remove these 8 variables from the xml files, this asks for a `genecec` modification which affects the  file_def xml files and the json data request files in the `r6874-control-output-files` branch and thus holds the `EC-Earth3.3.1.1` release.
We propose option 2, unless we expect active solving of several of these variables then I suggest to follow option 1.

**5. Emon nwdFracLut #469**
Can be deactivated in `genecec`, will need an update for the json data request files in the `r6874-control-output-files` branch and thus holds the `EC-Earth3.3.1.1` release. Can be done in within one day.

**6. wap_6hrPlev & zg_6hrPlevPt not passing PrePARE #462**
So skip, as it seems to be not in our own hands?:
https://github.com/PCMDI/cmor/issues/502

**7. Not ready - so skip?:#490**
IFS, open yet (zonal means, will probably be combined with #262 in `v1.3.0:`
```
ERROR:ece2cmor3.ifs2cmor: Variable zg has unsupported combination of dimensions [u'latitude', u'plev19', u'time'] and will be skipped.
ERROR:ece2cmor3.ifs2cmor: Variable ua has unsupported combination of dimensions [u'latitude', u'plev39', u'time'] and will be skipped.
ERROR:ece2cmor3.ifs2cmor: Variable ta has unsupported combination of dimensions [u'latitude', u'plev39', u'time'] and will be skipped.
```

Nemo, not investigated yet:
```
ERROR:ece2cmor3.nemo2cmor: Variable siareaacrossline     in table SImon      was not found in the NEMO output files: task skipped.
ERROR:ece2cmor3.nemo2cmor: Variable snmassacrossline     in table SImon      was not found in the NEMO output files: task skipped.  
```

**8. Implement nemo odec and oclim variables for cmip6 cmorization #262**
Doesn't hold any `EC-Earth3.3.1.1` release, and is assigned to ece2cmor3 v1.3.0, actually it is highly desirable to have it now in v1.2.0 but that will delay the release.

**Comment by plesager** [comment]
Thanks for the overview. Thomas has started to remove the problematic variables from the json data request file, aiming for a release tomorrow morning.

**Comment by goord** [comment]
The zg, ua and ta are skipped because they are zonal means. I don't think I will be able to fix add support for such zonal means by tomorrow.

**Comment by oloapinivad** [comment]
Thanks @treerink. 
I agree on 1 (keep as it is for now) and 4 (remove them from the json file). 
About 6 I still have no reply from PCDMI (I will inform you as soon as I get any feedback) but I agree that it is on their side. 
I also agree to skip 7 and 8. 

It would be nice to fix also 3 but I don't think we have enough time: I think this can be addressed in version 1.3.0 as well as the other warnings from QA-DKRZ #497

**Comment by treerink** [comment]
I have updated the first post: Point 4 and table, Point 7, the zonal means.

**Comment by oloapinivad** [comment]
Re. `hfcorr` and `wfcorr`, I think that they will be important in other type of experiments as DCPP ones where a flux correct is applied. 

In the OMIP paper (Griffies et al. 2016) it is stated that:
> hfcorr: This diagnostic records the heat flux correction acting at the liquid ocean surface. This field is zero for nearly all CMIP6 models. However, for the CMIP6/FAFMIP experiment (Sect. 3.2 and Gregory et al., 2016), the prescribed FAFMIP perturbation heat should be saved in the hfcorr diagnostic (see Table K3)
>  wfcorr: mass flux due to flux corrections. It is zero for models with no prescribed flux corrections/adjustments. However, for the CMIP6/FAFMIP experiment (Sect. 3.2), the prescribed perturbation water flux should be saved in the wfcorr diagnostic for
> FAFMIP models that make use of a real water flux.

I would drop them too at least in DECK runs

**Comment by treerink** [comment]
EC-Earth3.3.1.1 has been tagged: [https://svn.ec-earth.org/ecearth3/tags/3.3.1.1/ ](https://dev.ec-earth.org/projects/ecearth3/repository/revisions/6970/entry/ecearth3/tags/3.3.1.1) in EC-Earth3 r6970. This EC-Earth3 version will be the compatible version with `ece2cmor3` `v1.2.0`.

**Comment by treerink** [comment]
**Overview of DECK + historical + ScenarioMIP related issues:**

**1. ua100m & va100m #496**
For backward compatibility this issue won't be addressed in `ece2cmor3 v1.2.0`. However thereafter we will use grib codes `246.228` & `247.228` for `ua100m` & `va100m` as in the `ua-va-100m` branch is implemented already, running `genecec` with that will change the control output files.

**2. Ofx/sftof varies with time #483**
Because this is a time invariant variable which can be produced afterwards, we did not like to delay the `EC-Earth3.3.1.1 -- ece2cmor3 v1.2.0` compatible release by this issue. We should have taken off the `Ofx sftof` variable from the json data request file, however we forgot, so it will be cmorised but based on the wrong fields. Hopefully a `1.2.x` version will include a fix.

**3. Missing time bounds #472**
Appeared to be not a real issue.

**4. QA-DKRZ only invalid values #471**
These invalid-field variables have been removed from the json data request files, so won't show up among your cmorised output. So we followed approach 2 from [above](https://github.com/EC-Earth/ece2cmor3/issues/498#issue-457512428).

**5. Emon nwdFracLut #469**
Removed from the json data request files. Hopefully a `1.2.x` version will include a fix. Then `Emon nwdFracLut`needs to be included by `genecec` in the json data request files again.

**6. wap_6hrPlev & zg_6hrPlevPt not passing PrePARE #462**
Remains currently an open issue at [PCMDI](https://github.com/PCMDI/cmor/issues/502).

**7 & 8. Zonal means & climatologies #262 in `v1.3.0`**
See overview of skipped variables and ece2cmor log errors in #500.

**Comment by treerink** [comment]
Released:
https://github.com/EC-Earth/ece2cmor3/releases/tag/v1.2.0
